imap_data_access/io.py

```diff
@@ -159,6 +163,85 @@ def query(
             "At least one query parameter must be provided. "
             "Run 'query -h' for more information."
         )
+    # Check instrument name
+    if instrument is not None:
+        if instrument not in (
+            "codice",
+            "glows",
+            "hi",
+            "hit",
+            "idex",
+            "lo",
+            "mag",
+            "swapi",
+            "swe",
+            "ultra",
+        ):
+            raise ValueError(
+                "Not a valid instrument, please choose from "
+                "('codice', 'glows', 'hi', 'hit', 'idex', "
```

Comment:
Similarly, just print out the valid instruments here. Not that I think they're going to start bolting instruments on, but we have had this list change before